### ISSUE TITLE
Build DNA on aarch64 machine on Holo hosting app

### ISSUE THREAD

**Issue by rgouzal** [Holo-Host/holo-hosting-app-redux#56]
Hi I am trying to build DNA I have managed to compile and install holo hosting app on Raspberry Pi 4 running Ubuntu OS but I am having an issue when building the DNA with the below command:

`nix-build`

The error I am having:

```
while evaluating 'handleEvalIssue' at /nix/store/rfspa8qp1vyhpqn8jznq3hjb9rffgnq6-nixpkgs/pkgs/stdenv/generic/check-meta.nix:147:38, called from /nix/store/rfspa8qp1vyhpqn8jznq3hjb9rffgnq6-nixpkgs/pkgs/stdenv/generic/check-meta.nix:252:14:
Package ‘holo-hosting-app’ in /nix/store/lg7xbkvdif97xsn9g0xskkcg2vsgxfbc-source/overlays/holo-nixpkgs/build-dna/default.nix:26 is not supported on ‘aarch64-linux’, refusing to evaluate.
```

**Comment by alastairong** [comment]
Hmm I think the instructions are wrong. I usually build it with:

`nix-shell`
`make build`

Can you try that?

**Comment by rgouzal** [comment]
@alastairong I think it's building now thanks, but still getting another error:

```
> cargo build --release --target=wasm32-unknown-unknown
"cargo build --release --target=wasm32-unknown-unknown"
   Compiling provider v0.6.0 (/home/wowza/HOT/holo-hosting-app/zomes/provider/code)
error: linker `rust-lld` not found
  |
  = note: No such file or directory (os error 2)

error: aborting due to previous error

error: could not compile `provider`.
```

**Comment by alastairong** [comment]
Hmm I'm not sure to what extent we currently support compiling DNAs in aarch64. Is there any reason why you want to compile it on the Raspberry Pi rather than compiling it on x64 and then running that on the Raspberry Pi? The built files are platform agnostic.

**Comment by rgouzal** [comment]
@alastairong I am trying to see if it is possible to host an app using Raspberry Pi.

**Comment by alastairong** [comment]
Yes it should be possible - I have tested hosted hApps on other aarch64 devices. But we generally recommend that the aarch64 device download the compiled WASM from elsewhere. This is because if compiled locally, it will have a different hash and therefore be its own hashspace.

Also, I just realised this is the repo for the old version of HHA compatible with the now deprecated redux version of Holochain. The new version is not open source.

My recommendation for hosting on the raspberry pi is to install HPOS on the raspberry pi via the holo-nixpkgs repo. (Basically, this is installing NixOS and setting the profile to be the holoport profile).

Alternatively, the built HHA wasm can be found here:  https://holo-host.github.io/holo-hosting-app-rsm/releases/downloads/0_1_0_alpha21/core-app.0_1_0_alpha21.happ, but hosting requires other infrastructure components beyond the hApp alone.

**Comment by rgouzal** [comment]
@alastairong Pardon me I kinda new to this, what do you mean by HHA? and how can I run the happ file? is there any documentation?

**Comment by alastairong** [comment]
HHA  = holo-hosting app

There isn't any external documentation as the inner workings of Holo hosting is not open-source, unlike Holochain. Holo hosting isn't a single hApp. It is a system comprised of several hApps as well as non-hApp components both contained in HPOS and hosted centrally.

Our aim is to provide ways to set up HPOS and connect to the network, rather than to provide ways to deconstruct and reconstruct the entire system component-by-component.

If you want to participate in Holo hosting via a Raspberry Pi, you will need to run NixOS/HPOS on it, which will install all the necessary components. However, because we're still in alpha we're not allowing just anyone to start hosting yet. There's various reasons for this - e.g., lack of community support personnel, no smart hosting assignment so a raspberry pi could easily get overloaded with too many users.
